### PR TITLE
#13879 [Robosaur] Trigger ML assisted labeling when creating projects (PoC)

### PR DESCRIPTION
**Pull request by michael-jong-ds**
1. Added functions to trigger auto label (ML assisted labeling) on project creation

### CODE REVIEW

File: src/config/interfaces.ts [hunk 1]
```diff
@@ -220,6 +220,8 @@ export interface CreateConfig {
    * @description Optional. Tags to be applied to created projects
    */
   tagNames?: string[];
+
+  autoLabel: AutoLabelConfig;
```
Comment (jonathancesario) [inline src/config/interfaces.ts:224]: doesn't have to be? could be undefined

File: src/config/interfaces.ts [hunk 2]
```diff
@@ -387,3 +389,12 @@ export interface SplitDocumentConfig {
   filesPerFolder: number;
   resultFolder: string;
 }
+
+export interface AutoLabelConfig {
+  enableAutoLabel: boolean;
+  projectId: string;
+  labelerEmail: string;
```
Comment (jonathancesario) [inline src/config/interfaces.ts:396]: what is this for?

File: src/config/interfaces.ts [hunk 3]
```diff
@@ -387,3 +389,12 @@ export interface SplitDocumentConfig {
...
+  targetApiEndpoint: string;
+  targetApiSecretKey: string;
+  numberOfFilesPerRequest: number;
```
Comment (jonathancesario) [inline src/config/interfaces.ts:399]: is there any limitation for this attribute?

File: src/handlers/create-projects.handler.ts [hunk 4]
```diff
@@ -228,3 +236,58 @@ async function checkProjectCreationJob(
     await scriptState.save();
   }
 }
+
+function getProjectsToAutoLabel(projectsToBeCreated: { name: string; fullPath: string }[], scriptState: ScriptState) {
+  const newProjectNames = projectsToBeCreated.map((project) => project.name);
+  const filtered = new Map(
```
Comment (jonathancesario) [inline src/handlers/create-projects.handler.ts:242]: return right away?

**Review by jonathancesario — CHANGES_REQUESTED** [review]
how about creating a new file, e.g. `auto-label.handler.ts`? move all the logic and the create-projects handler could just import it

File: src/config/interfaces.ts (hunk 3 above)
Comment (michael-jong-ds) [reply src/config/interfaces.ts:399]: none that I know of, apa ada restriction dari api nya?

**Review by jonathancesario — APPROVED** [review]

**Pull request merged**